controllers/csr_controller.go

```diff
@@ -228,7 +232,18 @@ func (r *GuestCSRReconciler) getMachineAddresses(
 			}
 			providerID := "hcloud://bm-" + matches[2]
 			hList := &infrav1.HetznerBareMetalMachineList{}
-			if err := r.mCluster.List(ctx, hList, &client.ListOptions{}); err != nil {
+			selector := labels.NewSelector()
+			key := "cluster.x-k8s.io/cluster-name"
```

Comment:
there is a constant for this (clusterv1.ClusterLabel or something)

Reply:
@janiskemper, ok I use that constant now.